### ISSUE TITLE
Waydroid fails to start with [gbinder] ERROR: Can't open /dev/anbox-binder: No such file or directory

### ISSUE THREAD

**Issue by WarpspeedSCP** [waydroid/waydroid#262]
I look at /dev and cannot see anbox-binder; only anbox-hwbinder and anbox-vndbinder. Is there some line in the waydroid code I could change to solve this?

**Comment by Quackdoc** [comment]
could you run this `zgrep -i -e android -e ashmem /proc/config.gz` and post the output

**Comment by fortressia** [comment]
> could you run this `zgrep -i -e android -e ashmem /proc/config.gz` and post the output

```
# Android
# end of Android
# Android
# CONFIG_ANDROID is not set
# end of Android
```

**Comment by cloudw233** [comment]
# Android
CONFIG_ASHMEM=y
# end of Android
# Android
CONFIG_ANDROID=y
CONFIG_ANDROID_BINDER_IPC=y
CONFIG_ANDROID_BINDERFS=y
CONFIG_ANDROID_BINDER_DEVICES=""
CONFIG_ANDROID_BINDER_IPC_SELFTEST=y
# end of Android

**Comment by cloudw233** [comment]
this is the output

**Comment by cniw** [comment]
`CONFIG_ANDROID_BINDER_DEVICES=""`
from here we know that this kernel build config value is empty, so we need to assign the value through kernel boot flag

1. open default grub config file
    `sudo nano /etc/default/grub`
2. find line that contain `GRUB_CMDLINE_LINUX=` or `GRUB_CMDLINE_LINUX_DEFAULT=`
3. insert this flag `'binder.devices=binder,hwbinder,vndbinder,anbox-binder,anbox-hwbinder,anbox-vndbinder'` into one of them
4. save it 
5. update the grub config file
    `sudo grub-mkconfig -o /boot/grub/grub.cfg`
6. reboot, to make sure you're boot that flag
    `grep -o binder.devices=.* /proc/cmdline`

**Comment by aleasto** [comment]
Stale, missing info

**Comment by chkg2a** [comment]
> zgrep -i -e android -e ashmem /proc/config.gz

```
CONFIG_X86_ANDROID_TABLETS=m
# Android
# CONFIG_ANDROID_BINDER_IPC is not set`
# end of Android
```

I've inserted `binder.devices=binder,hwbinder,vndbinder,anbox-binder,anbox-hwbinder,anbox-vndbinder`  on my grub config and updated it. It doens't seem to do anything

> grep -o binder.devices=.* /proc/cmdline
```
zsh: no matches found: binder.devices=.*
```
> /boot/grub/grub.cfg
```
		linux	/vmlinuz-linux root=UUID=15701821-fd4e-4eef-bf4e-0176c1a83e58 rw rootfstype=ext4,binder.devices=binder,hwbinder,vndbinder,anbox-binder,anbox-hwbinder,anbox-vndbinder loglevel=3
```
> waydroid show-full-ui
```
[11:22:05] Starting waydroid session
^C^C[11:22:07] Failed to start Clipboard manager service, check logs
[gbinder] ERROR: Can't open /dev/anbox-binder: No such file or directory
[11:22:07] Failed to add presence handler: None
[gbinder] ERROR: Can't open /dev/anbox-binder: No such file or directory
[11:22:07] Failed to add presence handler: None
[gbinder] ERROR: Can't open /dev/anbox-binder: No such file or directory
[11:22:07] Failed to add presence handler: None
```
> ls /dev | grep anbox
```
anbox-hwbinder
anbox-vndbinder
```

I've installed anbox-modules-dkms-git, and was running fine. Checked a few days later and it stop working all together

Thank you

**Comment by baptx** [comment]
@cniw Thanks, it fixed the issue. But I did not need the single quotes since there were already double quotes.
I got the issue after disabling Waydroid autostart with the command `sudo systemctl disable waydroid-container` (otherwise Anbox was not working anymore):
https://github.com/anbox/anbox/issues/211#issuecomment-1869667654
Any idea, why Waydroid uses `/dev/anbox-binder` if Anbox is a separate project?
